Runner/JobManager.py

```diff
@@ -0,0 +1,332 @@
+        super(JobManager, self).__init__()
+        self.raw_job_id = ctm.get_job_id(parameters_file)
+        self.job_id = _time + "-" + self.raw_job_id
+        self.pool_id = ctm.get_pool_id(parameters_file)
+        self.template_file = template_file
+        self.parameters_file = parameters_file
+        self.application_licenses = application_licenses
+        self.expected_output = expected_output
+        self.pool_template_file = pool_template_file
+        self.storage_info = None
+        self.job_status = Utils.JobStatus(
+            Utils.JobState.NOT_STARTED,
+            "Job hasn't started yet.")
+        self.duration = None
+
+    def __str__(self) -> str:
+        return "job_id: [{}] pool_id: [{}] ".format(self.job_id, self.pool_id)
+
+    async def create_and_submit_Job(self, batch_service_client):
+        """
+        Creates the Job that will be submitted to the batch service
+
+        :param batch_client: The batch client to use.
+        :type batch_client: `batchserviceclient.BatchServiceClient`
+        """
+        Utils.logger.info('Creating Job [{}]... job will run on [{}]'.format(self.job_id, self.pool_id))
+
+        # load the template and parameters file
+        template = ctm.load_file(self.template_file)
+        parameters = ctm.load_file(self.parameters_file)
+
+        # overrides some of the parameters needed in the file, container SAS
+        # tokens need to be generated for the container
+        ctm.set_parameter_name(parameters, self.job_id)
+        ctm.set_parameter_storage_info(parameters, self.storage_info)
+
+        # Submits the job
+        await submit_job(batch_service_client, template, parameters)
+
+    async def create_pool(self, batch_service_client, image_references):
+        """
+        Creates the Pool that will be submitted to the batch service
+
+        :param batch_client: The batch client to use.
+        :type batch_client: `batchserviceclient.BatchServiceClient`
+        """
+        # load the template file
+        template = ctm.load_file(self.pool_template_file)
+
+        # set extra license if needed
+        if self.application_licenses is not None:
+            template["pool"]["applicationLicenses"] = self.application_licenses.split(
+                ",")
+
+        # Set rendering version
+        ctm.set_image_reference(template, image_references)
+
+        loop = asyncio.get_event_loop()
+        all_pools = [p.id for p in await loop.run_in_executor(None, functools.partial(batch_service_client.pool.list))]
+
+        ctm.set_template_name(template, self.pool_id)
+        if self.pool_id not in all_pools:
+            pool_json = batch_service_client.pool.expand_template(template)
+            pool = batch_service_client.pool.poolparameter_from_json(pool_json)
+            Utils.logger.info('Creating pool [{}]...'.format(self.pool_id))
+            try:
+                await loop.run_in_executor(None, functools.partial(batch_service_client.pool.add, pool))
```

Comment:
💡 I think it will be worth it to comment here, but pelase clarify is this funciton is `async` at all?